src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -37,6 +37,8 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):  # pyl
 # Arguments for 'iot hub policy' group
 register_cli_argument('iot hub policy', 'policy_name', options_list=('--name', '-n'), id_part='child_name',
                       help='Shared access policy name.')
+register_cli_argument('iot hub policy', 'permissions', help='Permissions of shared access policy.',
+                      **enum_choice_list(AccessRights))
```

Comment:
>  --permissions [Required]: Permissions of shared access policy.  Allowed values: 
> "RegistryRead, DeviceConnect", 
> "RegistryRead, RegistryWrite",
> "RegistryRead, RegistryWrite, DeviceConnect",
> "RegistryRead, RegistryWrite, ServiceConnect",
> "RegistryRead, RegistryWrite, ServiceConnect, DeviceConnect",
> "RegistryRead, ServiceConnect",
> "RegistryRead, ServiceConnect, DeviceConnect",
> "RegistryWrite, DeviceConnect",
> "RegistryWrite, ServiceConnect",
> "RegistryWrite, ServiceConnect, DeviceConnect",
> "ServiceConnect, DeviceConnect", 
> DeviceConnect,
> RegistryRead,
> RegistryWrite,
> ServiceConnect.

From the CLI perspective, the resulting "help" value (which I cleaned up for clarity) is basically unusable an results in an experience that diverges wildly from what is done elsewhere.

`... --permissions "RegistryRead, RegistryWrite"`

We don't accept comma-separated lists anywhere in the CLI, and while this isn't technically that (it's just a string containing spaces and commas) it's close enough that it will confuse users, who will be accustomed to using space-separated lists.

This enum essentially contains all possible combinations of these four permissions. I would argue that the underlying SDK enum should contain only the four basic values and a permissions object would track which permissions are set, rather than having an enum entry for every valid combination (especially since all combinations are valid).

Then the help text is useful:
>  --permissions [Required]: Space-separated list of permissions for the shared access policy.  Allowed values: 
>                DeviceConnect, RegistryRead, RegistryWrite, ServiceConnect.

And the usage is consistent with the rest of the CLI:
`... --permissions RegistryRead RegistryWrite ServiceConnect` 

Another acceptable pattern used elsewhere in the CLI would be to associate each permission with a single character and then parse the string (storage does this in places):

For example, assume the following mapping:
> DeviceConnect = d
> RegistryRead = r
> RegistryWrite = w
> ServiceConnect = s

Then the usage would look like:
`... --permissions srw` 

I would recommend the first approach (space-separated list of simple enum values) as it is clearer.